### PR TITLE
Druid code cleanup

### PR DESCRIPTION
**Pull request by sbelak**
Cleans up the code some and makes it compliant with our coding style.

### CODE REVIEW

File: modules/drivers/druid/test/metabase/driver/druid_test.clj [hunk 1]
```diff
@@ -202,13 +202,13 @@
 
 ;; count-where
 (expect-with-driver :druid
-  [[17.0]]
+  [[951]]
```
Comment (camsaul) [inline modules/drivers/druid/test/metabase/driver/druid_test.clj]: @sbelak I ran some queries locally and can confirm 951 is the right answer. 

How did we merge tests for the wrong answer (`17`) in the first place?

**Review by camsaul — APPROVED** [review]
This PR looks good, thanks for doing this, feel free to fix more stuff if you want

File: modules/drivers/druid/test/metabase/driver/druid_test.clj (hunk 1 above)
Comment (sbelak) [reply modules/drivers/druid/test/metabase/driver/druid_test.clj]: It's diffed off of sum-where branch ... which at some point had the wrong result

File: modules/drivers/druid/src/metabase/driver/druid/query_processor.clj [hunk 2]
```diff
@@ -84,41 +89,48 @@
...
   (du/date->iso-8601 (du/date-trunc unit time (get-timezone-id))))
 
-(defmethod ->rvalue :relative-datetime [[_ amount unit]]
+(defmethod ->rvalue :relative-datetime
+  [[_ amount unit]]
   (du/date->iso-8601 (du/date-trunc unit (du/relative-date unit amount) (get-timezone-id))))
 
-(defmethod ->rvalue :value [[_ value]]
+(defmethod ->rvalue :value
+  [[_ value]]
```
Comment (camsaul) [inline modules/drivers/druid/src/metabase/driver/druid/query_processor.clj]: I really don't have any complaint about putting arg lists on the same line for functions or methods without a docstring, any reason in particular to prefer always putting them on their own line?

File: modules/drivers/druid/test/metabase/driver/druid_test.clj (hunk 1 above)
Comment (camsaul) [reply modules/drivers/druid/test/metabase/driver/druid_test.clj]: ok

File: modules/drivers/druid/src/metabase/driver/druid/query_processor.clj (hunk 2 above)
Comment (sbelak) [reply modules/drivers/druid/src/metabase/driver/druid/query_processor.clj]: Not really except for consistency's sake (and multimethods with gnarly dispatch values are perhaps easier to glance)

**Pull request merged**